plugins/notion/src/blocksToHTML.ts

```diff
@@ -92,7 +92,38 @@ export function blocksToHtml(blocks: BlockObjectResponse[]) {
                 break
             }
             case "code":
-                htmlContent += `<pre><code class="language-${block.code.language.replace(" ", "-")}">${richTextToHTML(block.code.rich_text)}</code></pre>`
+                htmlContent += `<pre data-language="${
+                    CODE_LANGUAGE_MAPPING[block.code.language] || "JavaScript"
+                }"><code>${richTextToHTML(block.code.rich_text)}</code></pre>`
+                break
+            case "quote":
+                htmlContent += `<blockquote>${richTextToHTML(block.quote.rich_text)}</blockquote>`
+                break
+            case "video":
+                if (block.video.type === "external") {
+                    const url = block.video.external.url
+                    if (url && (url.includes("youtube.com") || url.includes("youtu.be"))) {
+                        try {
+                            const urlObj = new URL(url)
+                            let videoId = ""
+
+                            if (urlObj.hostname === "youtube.com" || urlObj.hostname === "www.youtube.com") {
+                                videoId = urlObj.searchParams.get("v") || ""
+                            } else if (urlObj.hostname === "youtu.be") {
+                                videoId = urlObj.pathname.slice(1)
+                            }
+
+                            if (videoId && /^[a-zA-Z0-9_-]{11}$/.test(videoId)) {
+                                const embedUrl = `https://www.youtube.com/embed/${encodeURIComponent(
+                                    videoId
+                                )}?controls=0&autoplay=0&loop=0&mute=1`
+                                htmlContent += `<iframe src="${embedUrl}"></iframe>`
+                            }
+                        } catch (error) {
+                            console.warn("Invalid YouTube URL:", url)
+                        }
```

Comment:
We could potentially rely on RegEx to simplify this logic:

```tsx
const VIMEO_ID_REGEX = /vimeo\.com\/(?<videoId>[^?]+)/u
const YOUTUBE_ID_REGEX = /(?:youtu\.be\/|youtube\.com\/(?:watch\?v=|embed\/))(?<videoId>[^?&]+)/u
```
I'll let you adjust the embed src for Vimeo, or you can use Framer's default by not passing any query variables.
```suggestion
 const url = block.video.external.url

                    const vimeoId = url.match(VIMEO_ID_REGEX)?.groups?.videoId
                    const youtubeId = url.match(YOUTUBE_ID_REGEX)?.groups?.videoId

                    if (vimeoId) {
                        htmlContent += `<iframe src="https://player.vimeo.com/video/${vimeoId}"></iframe>`
                        break
                    } else if (youtubeId) {
                        htmlContent += `<iframe src="https://www.youtube.com/embed/${youtubeId}"></iframe>`
                        break
                    }

                    console.warn("Unsupported video URL:", block.video.external.url)
```

Reply:
Great, this is much simpler. I updated it to use your code for YouTube videos, but left Vimeo videos out because it shows this in the formatted text when importing videos from Vimeo:

<img width="717" alt="image" src="https://github.com/user-attachments/assets/cc52df2b-cf47-47c2-884a-804cb922bb03">

Since Vimeo videos aren't supported in the formatted text editor yet, we can leave it out for now and add it later when it becomes officially supported.